### DISCUSSION TITLE
remote synchronizing and local idle makes no sense

### DISCUSSION THREAD

**Discussion by muzzol** [Martchus/syncthingtray#273] [General]
I'm not sure if this belongs here or if it belongs to syncthing main project but I'm trying to scripting some actions after syncronizing a dir and I've found that local and remote machines can be in different states even when the sync is running.

how can two machines be synchronizing and one of them be idle?

I'm running the exact command on both:

```
syncthingctl -s -d id-XXXXX

```
and the remote one shows 

```
   Status                        synchronizing
      Global                        5965 file(s), 597 dir(s), 48.52 GiB
      Local                         5955 file(s), 597 dir(s), 43.32 GiB

```

and the local one shows

```
   Status                        idle
      Global                        5965 file(s), 597 dir(s), 48.52 GiB
      Local                         5965 file(s), 597 dir(s), 48.52 GiB

```

If I shutdown local machine the sync obviously stops so how can be idle?

On my setup I have no **central** machine, I have some desktops placed around home, work and other sites so I don't know when one is up or down, and the only one I'm checking for ending the transfer is the local one.
local files are still being transmited and they are still physically only on local machine, so it makes absolutely no sense to me to show it as idle.

Am I missing something here? 

thanks for your time.

**Comment by Martchus** [comment]
> even when the sync is running.

There is not "the" sync. Every device/folder is trying to get up-to-date individually by pulling data from whatever other device can serve them.

> local files are still being transmited and they are still physically only on local machine, so it makes absolutely no sense to me to show it as idle.

Maybe the wording chosen by Syncthing itself and also how Syncthing Tray displays the status is not perfect. Let me try to explain the wording:

* The status "synchronizing" means "pulling data from other devices to get up-to-date again".
* The status "idle" means "up-to-date". The official web UI actually displays it as "Up to Date"/"Aktuell" instead of "idle" (which is still the internal name of the status and Syncthing Tray displays that directly as I have never bothered to "translate" it to something that makes more sense).

With these definitions in mind it hopefully makes sense that the directory that is already up-to-date is shown as "idle"/"Up to Date".

I guess I should change Syncthing Tray to also use "Up to Date" and maybe generally streamline the wording with the official web UI.

Note that Syncthing does not have a separate status for "serving data for other device". In Syncthing Tray you can however configure that the state of connected remote devices is considered for the overall status (but not for the status of individual folders).

**Reply by muzzol** [reply]
I understand that the wording is imprecise but in any case when local device is doing something (giving or receiving data) it shouldn't be Idle.

I'm really surprised that there's not other people complaining about this.

my use case is really simple: you put a 10G movie in one synced directory and you want to poweroff the computer once is fully transfered. there's simply no corner case here, local device NEED to be up to finish the sync.

> Note that Syncthing does not have a separate status for "serving data for other device"

I guess this is the root problem, maybe I'll create a new ticket for that on syncthing project. are you in touch with them?

**Reply by Martchus** [reply]
I don't think they'll change the meaning of "idle" or introduce a new status because that would probably break compatibility or add unwanted complexity - considering you can simply use the [API to query the completion of a folder on a remote device](https://docs.syncthing.net/rest/db-completion-get.html) today. This is of course not as simple as querying a single status value but does what you need.

This is also what the official UI and Syncthing Tray do for displaying the remote completion (and the mentioned feature of Syncthing Tray to incorporate the remote completion into the overall status).

Unfortunately `syncthingctl` does not take the remote completion into account yet. So e.g. `syncthingctl wait-for-idle --timeout 1800000 --at-least 5000 && systemctl poweroff` will only wait until the folder is locally up-to-date. It is conceivable to add a `--remote` flag to change that but it hasn't been implemented yet. In the meantime you can simply let `syncthingctl` connect to the remote device via `--url` or write your own script using the mentioned API (or contribute the missing `--remote` flag to `syncthingctl`).

**Reply by Martchus** [reply]
> are you in touch with them?

Not really. I'm sometimes active on https://forum.syncthing.net which is also a good place for questions like that. (And I think there's also already an existing thread about this use case somewhere on that forum.)

**Reply by Martchus** [reply]
Actually, on second though, I think `syncthingctl wait-for-idle …` will actually wait for remote completion is you specify a device. It'll then wait until that device (all its folders) is completely considered up-to-date. Just the dirs specified via `--dir` are local-only.

**Reply by muzzol** [reply]
> Actually, on second though, I think `syncthingctl wait-for-idle …` will actually wait for remote completion is you specify a device. It'll then wait until that device (all its folders) is completely considered up-to-date. Just the dirs specified via `--dir` are local-only.

ok, I'll try that. thanks for your detailed explanation
